crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -1817,37 +1817,42 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
         binding_type(self.db(), class_definition).into_class_literal()
     }
 
-    /// Returns `true` if the current scope is the function body scope of a function overload (that
-    /// is, the stub declaration decorated with `@overload`, not the implementation), or an
-    /// abstract method (decorated with `@abstractmethod`.)
-    fn in_function_overload_or_abstractmethod(&self) -> bool {
+    /// If the current scope is a method, return the decorators applied to the method.
+    ///
+    /// If the current scope is a function, return `None`.
+    fn current_method_decorators(&self) -> Option<Vec<&ast::Decorator>> {
```

Comment:
I think it would be a bit nicer if this method instead returned a `FunctionDecorators` bitflag, rather than allocating a vector and requiring callers to sometimes clone it, and do all the type inference etc themselves.

Extra good would be if we could consolidate with the code that currently lives in `TypeInferenceBuilder::infer_function_definition` so we just have one occurrence of the match statement to build a `FunctionDecorators` bitflag.